azurerm/internal/services/network/resource_arm_virtual_hub_connection.go

```diff
@@ -0,0 +1,304 @@
+			return tf.ImportAsExistsError("azurerm_virtual_hub_connection", *connection.ID)
+		}
+	}
+
+	connection := network.HubVirtualNetworkConnection{
+		Name: utils.String(name),
+		HubVirtualNetworkConnectionProperties: &network.HubVirtualNetworkConnectionProperties{
+			RemoteVirtualNetwork: &network.SubResource{
+				ID: utils.String(d.Get("remote_virtual_network_id").(string)),
+			},
+			AllowHubToRemoteVnetTransit:         utils.Bool(d.Get("allow_hub_to_remote_vnet_transit").(bool)),
+			AllowRemoteVnetToUseHubVnetGateways: utils.Bool(d.Get("allow_remote_vnet_to_use_hub_vnet_gateways").(bool)),
+			EnableInternetSecurity:              utils.Bool(d.Get("enable_internet_security").(bool)),
+		},
+	}
+	connections = append(connections, connection)
+	virtualHub.VirtualHubProperties.VirtualNetworkConnections = &connections
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, virtualHubName, virtualHub)
+	if err != nil {
+		return fmt.Errorf("Error adding Connection %q to Virtual Hub %q (Resource Group %q): %+v", name, virtualHubName, resourceGroup, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for addition of Connection %q to Virtual Hub %q (Resource Group %q): %+v", name, virtualHubName, resourceGroup, err)
+	}
+
+	virtualHub, err = client.Get(ctx, resourceGroup, virtualHubName)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): %+v", virtualHubName, resourceGroup, err)
+	}
+	if virtualHub.VirtualHubProperties == nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): `properties` was nil", virtualHubName, resourceGroup)
+	}
+	if virtualHub.VirtualHubProperties.VirtualNetworkConnections == nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): `properties.VirtualNetworkConnections` was nil", virtualHubName, resourceGroup)
+	}
+
+	connections = *virtualHub.VirtualHubProperties.VirtualNetworkConnections
+	newConnection, _ := findVirtualHubConnection(name, connections)
+	if newConnection == nil {
+		return fmt.Errorf("Connection %q was not found in Virtual Hub %q / Resource Group %q", name, virtualHubName, resourceGroup)
+	}
+	if newConnection.ID == nil {
+		return fmt.Errorf("Error retrieving Connection %q (Virtual Hub %q / Resource Group %q): `id` was nil", name, virtualHubName, resourceGroup)
+	}
+
+	d.SetId(*newConnection.ID)
+	return resourceArmVirtualHubConnectionRead(d, meta)
+}
+
+func resourceArmVirtualHubConnectionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.VirtualHubClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ParseVirtualHubConnectionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	virtualHubName := id.VirtualHubName
+	name := id.Name
```

Comment:
I don't think we need to do this? eg id.ResourceGroup can just be used instead

Reply:
Updated